src/models/Observation.js

```diff
@@ -313,19 +315,30 @@ class Observation extends Realm.Object {
     }
   }
 
+  static filterUnsyncedObservations = realm => {
+    const unsyncedFilter = "_synced_at == null || _synced_at <= _updated_at";
+    const photosUnsyncedFilter = "ANY observationPhotos._synced_at == null";
+
+    const obs = realm?.objects( "Observation" );
+    const unsyncedObs = obs.filtered( `${unsyncedFilter} || ${photosUnsyncedFilter}` );
+    return unsyncedObs;
+  }
+
+  static isUnsyncedObservation = ( realm, obs ) => {
+    const obsList = Observation.filterUnsyncedObservations( realm );
+    const unsyncedObs = obsList.filtered( `uuid == "${obs.uuid}"` );
+    return unsyncedObs.length > 0;
+  }
+
```

Comment:
These are the kinds of things I think belong in the model, though I don't think we need to specify the model name in the method name unless the method is returning some other kind of data you might not expect, so `unsynced()` and `isUnsynced()` would probably suffice. If you're trying to start methods names with verbs, maybe just `filterUnsynced()` or `getUnsynced()`.

`isUnsynced` feels more like an instance method, though. Maybe something like

```javascript
isUnsynced( ) {
  return (
    this._synced_at == null
    || this._synced_at <= this._updated_at
    || this.observationPhotos.filtered( "ANY _synced_at == null" ).length > 0
  );
}
```

Reply:
Is this another thing we can audit as part of the tech check? I don't think we're using instance methods anywhere in the models, and I think I'd like to wrap my head around the difference between those and static functions more before making changes.

Reply:
You mean like make an issue like, "use instance methods where possible"? I can do that. In general, object-oriented design says that if an operation only pertains to a single instance of an object, like `isUnsynced` does here, it should be an instance method, i.e. the instance is best-suited to perform that operation b/c it should know everything about itself. This kind of [encapsulation](https://en.wikipedia.org/wiki/Object-oriented_programming#Encapsulation) often makes code more reusable, but it also makes code more understandable and easier to write, e.g. if you know you have an instance of an Observation you can think to yourself, "whose job is it to know if this observation needs to be synced? Probably the observation itself." If you don't have that expectation code becomes much harder to write and manage, e.g. "Is there an ObservationSyncChecker that I need to get to check if this observation is synced?"

You might use a static / class method when you want to perform an operation that involves multiple instances of that class, or that operates on instances of many classes, e.g. a `getUnsynced()` method that gets all unsynced observations.